tools/build_defs/pkg/make_deb.py

```diff
@@ -159,7 +159,8 @@ def CreateDebControl(extrafiles=None, **kwargs):
   with gzip.GzipFile('control.tar.gz', mode='w', fileobj=tar, mtime=0) as gz:
     with tarfile.open('control.tar.gz', mode='w', fileobj=gz) as f:
       tarinfo = tarfile.TarInfo('control')
-      tarinfo.size = len(controlfile)
+      # Don't discard unicode characters when computing the size
+      tarinfo.size = len(controlfile.encode('utf-8'))
```

Comment:
You might as well do the encoding just once.

Reply:
I agree. However, I get a TypeError from the `f.addfile(... BytesIO(controlfile))` line if I do the encoding at the top of the method. It will work if I do:
```python
controlfile = ''
  for values in DEBIAN_FIELDS:
    fieldname = values[0]
    key = fieldname[0].lower() + fieldname[1:].replace('-', '')
    if values[1] or (key in kwargs and kwargs[key]):
      controlfile += MakeDebianControlField(fieldname, kwargs[key], values[2])
  controlfile = controlfile.encode('utf-8')
...
```
and remove the encode() calls further down.

Do you have any idea why that would be? My guess would be the `controlfile += MakeDebianControlField(...)` line in the `for` loop but I'm unsure why that would override the encoding if it's set earlier. It seems like both this TypeError and the issues I run into `MakeDebianControlField` function result from the `installed_size` flag.